### PR TITLE
[openwebnet] Thermo: added support for HOLIDAY/VACATION central unit modes

### LINKED ISSUE #16539: [openwebnet] Thermo: add support for HOLIDAY/VACATION modes visualization for Central Unit
Currently it's not possibile to get in the `mode` channel when the active mode for the Thermo Central Unit is `HOLIDAY` or `VACATION`.
It would be useful to support also these modes when they are activated from the Central Unit control panel.

### PR DESCRIPTION
**Pull request by mvalla**
This PR adds support for `HOLIDAY` / `VACATION` Thermo Central Unit modes.
Fixes #16539.

Currently it's not possibile in `mode` channel to recognize when the active mode for the Thermo Central Unit is `HOLIDAY` or `VACATION` nor is possible to set these modes from openHAB.
This PR add support for these 2 modes, adding a new channel: `vacationDays` to set the number of vacation days on the Thermo central unit.

This PR also bumps `openwebnet4j` lib to version 0.13.0 with complete re-factored Thermo mode handling.

### CODE REVIEW

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java [hunk 1]
```diff
@@ -163,18 +170,20 @@ protected String ownIdPrefix() {
...
+            try {
+                Thermoregulation.FanCoilSpeed speed = Thermoregulation.FanCoilSpeed.valueOf(command.toString());
+                send(Thermoregulation.requestWriteFanCoilSpeed(w.value(), speed));
+            } catch (OWNException e) {
+                logger.warn("handleSetFanSpeed() {}", e.getMessage());
+            } catch (IllegalArgumentException e) {
+                logger.warn("handleSetFanSpeed() Unsupported command {} for thing {}", command, getThing().getUID());
+                return;
+            }
+
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:186]: Please remove this new blank line (you should see a warning when building).

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java [hunk 2]
```diff
@@ -221,26 +240,68 @@ private void handleSetProgramNumber(ChannelUID channel, Command command) {
...
                 }
+            } else {
+                newTemp = ((DecimalType) command).doubleValue();
+            }
+            try {
+                send(Thermoregulation.requestWriteSetpointTemperature(getWhere(w.value()), newTemp, currentFunction));
+            } catch (MalformedFrameException | OWNException e) {
+                logger.warn("handleSetpoint() {}", e.getMessage());
             }
+
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:304]: Please remove this new blank line (you should see a warning when building).

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java [hunk 3]
```diff
@@ -298,16 +388,18 @@ private String getWhere(String where) {
...
-                    Thermoregulation.Function function = Thermoregulation.Function.valueOf(command.toString());
-                    send(Thermoregulation.requestWriteFunction(w.value(), function));
-                } catch (OWNException e) {
-                    logger.warn("handleFunction() {}", e.getMessage());
-                } catch (IllegalArgumentException e) {
-                    logger.warn("handleFunction() Unsupported command {} for thing {}", command, getThing().getUID());
-                    return;
-                }
+            if (w == null) {
+                logger.warn("handleVacationDays() Where is null for thing {}", getThing().getUID());
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:392]: Method is handleFunction

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java [hunk 4]
```diff
@@ -399,65 +491,85 @@ protected void handleMessage(BaseOpenMessage msg) {
...
     private void updateModeAndFunction(Thermoregulation tmsg) {
         if (tmsg.getWhat() == null) {
-            logger.warn("updateModeAndFunction() Could not parse Mode or Function from {} (WHAT is null)",
-                    tmsg.getFrameValue());
+            logger.warn("updateModeAndFunction() Could not parse What {} (WHAT is null)", tmsg.getFrameValue());
             return;
         }
-        Thermoregulation.WhatThermo w = Thermoregulation.WhatThermo.fromValue(tmsg.getWhat().value());
-        if (w.getMode() == null) {
+        Thermoregulation.WhatThermo what = tmsg.new WhatThermo(tmsg.getWhat().value());
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:497]: I don't know this way to call a constructor. For my information, what does exactly `tmsg.new WhatThermo` ?

Comment (mvalla) [reply bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java:497]: It's the right way to create a new instance of an inner class. 
Did not know that, but Eclipse highlights a compilation error if i use `new Thermoregulation.WhatThermo()`:

_No enclosing instance of type Thermoregulation is accessible. Must qualify the allocation with an enclosing instance of type Thermoregulation (e.g. x.new A() where x is an instance of Thermoregulation)._

**Comment by mvalla** [comment]
@lolodomo I fixed all requested changes!

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**